### ISSUE TITLE
Test failures

### ISSUE THREAD

**Issue by whart222** [conda-forge/pyomo.extras-feedstock#1]
There are two weird test failures for pyomo.extras:

1. The TravisCI failure is unexpected since we just approved the PR for this feedstock and all tests passed, and
2. The Appveyor badge isn't working properly 

@pelson Can you help diagnose the issue?  Note that this is a pseudo-package, that only references other packages.  That might be part of the problem.

**Comment by jakirkham** [comment]
I don't think so. Based on the logs it appears the issue occurs during upload. So, it appears the tokens did not get setup correctly.

**Comment by pelson** [comment]
`
# $ anaconda upload /Users/travis/miniconda3/conda-bld/noarch/pyomo.extras-3.0-py_0.tar.bz2
#
...
$ ./ci_support/upload_or_check_non_existence.py ./recipe conda-forge --channel=main

[BinstarError] file /Users/travis/miniconda3/conda-bld/osx-64/pyomo.extras-3.0-py_0.tar.bz2 does not exist
`

Can you spot it? The upload script doesn't currently support noarch packages and so is baulking because it can't find the package in the `osx-64` directory.

We have thus far avoided noarch packages, simply because they are hard to verify on all platforms when built on only one. There are good cases for noarch packages, but IMO they are quite limited, and often only limited to metapackages which don't require extensive testing (because we can't test them on all platforms anyway).

Basically, we can start looking at adding noarch capability, but the fastest solution for you right now @whart222 would be to remove the noarch declaration in the recipe.

**Comment by whart222** [comment]
Pyomo.extras is a metapackage.  It simply pulls in other packages.  Hence,
there us a minimal need for testing.

Does conda-forge support metapackage like this?  I could change the
configuration to specify architectures if that's necessary.
On May 30, 2016 1:34 AM, "Phil Elson" <notifications@github.com> wrote:

`
$ anaconda upload
/Users/travis/miniconda3/conda-bld/noarch/pyomo.extras-3.0-py_0.tar.bz2

#
...
$ ./ci_support/upload_or_check_non_existence.py ./recipe conda-forge
--channel=main

[BinstarError] file
/Users/travis/miniconda3/conda-bld/osx-64/pyomo.extras-3.0-py_0.tar.bz2
does not exist
`

Can you spot it? The upload script doesn't currently support noarch
packages and so is baulking because it can't find the package in the osx-64
directory.

We have thus far avoided noarch packages, simply because they are hard to
verify on all platforms when built on only one. There are good cases for
noarch packages, but IMO they are quite limited, and often only limited to
metapackages which don't require extensive testing (because we can't test
them on all platforms anyway).

Basically, we can start looking at adding noarch capability, but the
fastest solution for you right now @whart222 <https://github.com/whart222>
would be to remove the noarch declaration in the recipe.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/conda-forge/pyomo.extras-feedstock/issues/1#issuecomment-222414682>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAsb-CFrEjRP0a8X92cmQlv0k3uWotmNks5qGnblgaJpZM4IpEth>
.

**Comment by jakirkham** [comment]
> Can you spot it? The upload script doesn't currently support noarch packages and so is baulking because it can't find the package in the `osx-64` directory.

Apparently not. 😆

> Does conda-forge support metapackage like this? I could change the configuration to specify architectures if that's necessary.

As @pelson said, no, it doesn't. Though that is easily fixed as done in this PR ( https://github.com/conda-forge/pyomo.extras-feedstock/pull/2 ).
